midealocal/devices/e2/__init__.py

```diff
@@ -74,12 +81,22 @@ def __init__(
                 DeviceAttributes.heating_power: None,
             },
         )
-        self._default_old_protocol = "auto"
+        self._default_old_protocol = OldProtocol.auto
         self._old_protocol = self._default_old_protocol
         self.set_customize(customize)
 
-    def old_protocol(self) -> bool:
-        return self.subtype <= 82 or self.subtype == 85 or self.subtype == 36353
+    def _normalize_old_protocol(self, value: Any) -> OldProtocol:
+        try:
+            if isinstance(value, str):
+                value = OldProtocol(value)
+            elif isinstance(value, int):
+                value = OldProtocol.true if value else OldProtocol.false
+            if not isinstance(value, OldProtocol):

```

Comment:
Better remove unneeded comparison
```suggestion
            else:
```